config/spotbugs/exclude.xml

```diff
@@ -14,251 +14,194 @@
-        </Not>
-    </Match>
-
-    <!-- Deliberate use of an unused field in the Person POJO class -->
-    <Match>
-        <Class name="com.mongodb.acceptancetest.crud.pojo.Person"/>
-        <Bug pattern="SS_SHOULD_BE_STATIC"/>
-    </Match>
-
-    <!-- Spock tests seem to fail the serial version ID test when stubbing.  Annoying. -->
-    <Match>
-        <Source name="~.*\.groovy"/>
-        <Bug pattern="SE_NO_SERIALVERSIONID,LI_LAZY_INIT_STATIC,EQ_UNUSUAL,IT_NO_SUCH_ELEMENT,RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT"/>
-    </Match>
-
-    <!-- Strange findbugs issues -->
-    <Match>
-        <Source name="~.*ClusterSettingsSpecification\.groovy"/>
-        <Bug pattern="RANGE_ARRAY_INDEX"/>
-    </Match>
-
-    <Match>
-        <Class name="~com.mongodb.internal.connection.AbstractByteBufBsonDocument.*"/>
-        <Bug pattern="NP_BOOLEAN_RETURN_NULL"/>
-    </Match>
-
-    <Match>
+        <!-- MongoDB status: "No Fix Needed", SpotBugs rank: 16 -->
         <Class name="~com.mongodb.client.ClientSession.*"/>
         <Bug pattern="NM_SAME_SIMPLE_NAME_AS_INTERFACE"/>
     </Match>
 
     <Match>
+        <!-- MongoDB status: "No Fix Needed", SpotBugs rank: 16 -->
         <Class name="~com.mongodb.reactivestreams.client.ClientSession.*"/>
         <Bug pattern="NM_SAME_SIMPLE_NAME_AS_INTERFACE"/>
     </Match>
 
     <Match>
-        <Class name="~com.mongodb.kotlin.client.ClientSession.*"/>
-        <Bug pattern="NM_SAME_SIMPLE_NAME_AS_INTERFACE"/>
-    </Match>
-
-    <Match>
+        <!-- MongoDB status: "No Fix Needed", SpotBugs rank: 16 -->
         <Class name="~com.mongodb.kotlin.client.coroutine.ClientSession.*"/>
         <Bug pattern="NM_SAME_SIMPLE_NAME_AS_INTERFACE"/>
     </Match>
 
     <Match>
-        <Class name="~com.mongodb.internal.async.client.ClientSession.*"/>
-        <Bug pattern="NM_SAME_SIMPLE_NAME_AS_INTERFACE"/>
-    </Match>
-
-    <Match>
-        <Class name="~com.mongodb.async.client.internal.AsyncCryptConnection.*"/>
-        <Bug pattern="RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT"/>
-    </Match>
-
-    <!-- Spotbugs assumes that SSLSocket#getSSLParameters never returns null, when that is not the case for all JDKs -->
-    <Match>
+        <!-- MongoDB status: "No Fix Needed", SpotBugs rank: 18 -->
+        <!-- Spotbugs assumes that SSLSocket#getSSLParameters never returns null, when that is not the case for all JDKs -->
         <Class name="com.mongodb.internal.connection.SslHelper"/>
         <Bug pattern="RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE"/>
     </Match>
 
     <Match>
+        <!-- MongoDB status: "False Positive", SpotBugs rank: 17 -->
         <Class name="com.mongodb.client.internal.CryptConnection"/>
         <Method name="retain"/>
         <Bug pattern="RV_RETURN_VALUE_IGNORED_NO_SIDE_EFFECT"/>
     </Match>
 
     <Match>
-        <Class name="~com.mongodb.reactivestreams.client.internal.crypt.*"/>
+        <!-- MongoDB status: "False Positive", SpotBugs rank: 17 -->
```

Comment:
`No Fix Needed` seems better.

Reply:
I think this is actually "False Positive", because the finding description is _"This code calls a method and ignores the return value. However our analysis shows that the method (including its implementations in subclasses if any) **does not produce any effect other than return value**. Thus this call can be removed."_ The highlighted part is false.